packages/toolpad-app/src/toolpad/AppEditor/HierarchyExplorer/CreateCodeComponentNodeDialog.tsx

```diff
@@ -101,10 +101,8 @@ export default function CreateCodeComponentDialog({
           const appNode = appDom.getApp(dom);
 
           appStateApi.update((draft) => appDom.addNode(draft, newNode, appNode, 'codeComponents'), {
```

Comment:
Not introduced in this PR, but is there any technical reason why `update` updates both the dom and the view? It feels like two different things to me

Reply:
In some cases for the undo/redo history to be good the best is that we update the DOM and the view in the same action, especially when adding or removing nodes. Maybe there's a better alternative though?